controllers/osbuild_controller.go

```diff
@@ -47,16 +81,220 @@ type OSBuildReconciler struct {
 }
 
+func (r *OSBuildReconciler) UpdateContainerComposeId(ctx context.Context, logger logr.Logger, osBuild *osbuildv1alpha1.OSBuild) (composer.ComposeStatusValue, error) {
+	composeStatus, err := r.checkComposeIDStatus(ctx, logger, osBuild.Status.ContainerComposeId)
+	if err != nil {
+		logger.Error(err, "failed to get compose ID status")
+		return "", err
+	}
+
+	r.updateOSBuildConditionStatus(ctx, logger, osBuild, composeStatus, ContainerBuildDone, FailedContainerBuild, StartedContainerBuild)
+	return composeStatus, nil
+}
+
+func (r *OSBuildReconciler) UpdateIsoComposeId(ctx context.Context, logger logr.Logger, osBuild *osbuildv1alpha1.OSBuild) (composer.ComposeStatusValue, error) {
+	composeStatus, err := r.checkComposeIDStatus(ctx, logger, osBuild.Status.IsoComposeId)
+	if err != nil {
+		logger.Error(err, "failed to get compose ID status")
+		return "", err
+	}
+
+	r.updateOSBuildConditionStatus(ctx, logger, osBuild, composeStatus, IsoBuildDone, FailedIsoBuild, StartedIsoBuild)
+	return composeStatus, nil
+}
+
+func (r *OSBuildReconciler) updateOSBuildConditionStatus(ctx context.Context, logger logr.Logger,
+	osBuild *osbuildv1alpha1.OSBuild, composeStatus composer.ComposeStatusValue,
+	buildDoneValue osbuildv1alpha1.OSBuildConditionType, buildFailedValue osbuildv1alpha1.OSBuildConditionType,
+	buildStartedValue osbuildv1alpha1.OSBuildConditionType) {
+
+	if composeStatus == composer.ComposeStatusValueSuccess {
+		r.updateOSBuildStatus(ctx, logger, osBuild, EdgeContainerJobFinishedMsg, buildDoneValue, EmptyComposeID, EmptyComposeID)
+	}
+
+	if composeStatus == composer.ComposeStatusValueFailure {
+		r.updateOSBuildStatus(ctx, logger, osBuild, EdgeContainerJobFailedMsg, buildFailedValue, EmptyComposeID, EmptyComposeID)
+	}
+	if composeStatus == composer.ComposeStatusValuePending {
+		r.updateOSBuildStatus(ctx, logger, osBuild, EdgeContainerJobStillRunningMsg, buildStartedValue, EmptyComposeID, EmptyComposeID)
+	}
+
+}
+
+func (r *OSBuildReconciler) postComposeEdgeContainer(ctx context.Context, logger logr.Logger, osBuild *osbuildv1alpha1.OSBuild) error {
+	customizations := r.createCustomizations(*osBuild.Spec.Details.Customizations)
+	imageRequest := r.createImageRequest(osBuild.Spec.Details.TargetImage, EdgeContainerImgType)
+
+	body := composer.PostComposeJSONRequestBody{
+		Customizations: &customizations,
+		Distribution:   osBuild.Spec.Details.Distribution,
+		ImageRequest:   &imageRequest,
+	}
+
+	// post compos:
+	response, err := r.ComposerClient.PostCompose(ctx, body)
+	if err != nil {
+		logger.Error(err, "failed to post a new request")
+		r.updateOSBuildStatus(ctx, logger, osBuild, FailedToSendPostRequestMsg, FailedContainerBuild, EmptyComposeID, EmptyComposeID)
+		return err
+	}
+
+	composerResponse, err := composer.ParsePostComposeResponse(response)
+	if err != nil {
+		logger.Error(err, "failed parsing the response of postCompose")
+		return err
+	}
+	if composerResponse.StatusCode() != http.StatusCreated {
```

Comment:
If the composer returned an error, this means that the Container build failed. So, the resource Status needs to be updated as well.